File: weblayer/shell/android/browsertests_apk/translate_test_bridge.cc
// Copyright 2020 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "weblayer/shell/android/browsertests_apk/translate_test_bridge.h"

#include "base/android/jni_android.h"
#include "weblayer/test/weblayer_browsertests_jni/TranslateTestBridge_jni.h"

using base::android::JavaParamRef;
using base::android::ScopedJavaLocalRef;

namespace weblayer {

// static
void TranslateTestBridge::SelectButton(
    TranslateCompactInfoBar* infobar,
    TranslateCompactInfoBar::ActionType action_type) {
  JNIEnv* env = base::android::AttachCurrentThread();
  Java_TranslateTestBridge_selectTab(env, infobar->GetJavaInfoBar(),
                                     action_type);
}

// static
void TranslateTestBridge::ClickOverflowMenuItem(
    TranslateCompactInfoBar* infobar,
    OverflowMenuItemId item_id) {
  JNIEnv* env = base::android::AttachCurrentThread();
  switch (item_id) {
    case OverflowMenuItemId::NEVER_TRANSLATE_LANGUAGE:
      Java_TranslateTestBridge_clickNeverTranslateLanguageMenuItem(
          env, infobar->GetJavaInfoBar());
      return;
    case OverflowMenuItemId::NEVER_TRANSLATE_SITE:
      Java_TranslateTestBridge_clickNeverTranslateSiteMenuItem(
          env, infobar->GetJavaInfoBar());
      return;
  }
}

}  // namespace weblayer

File: weblayer/shell/android/browsertests_apk/src/org/chromium/weblayer_private/TranslateTestBridge.java
// Copyright 2020 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.weblayer_private;

import org.chromium.base.annotations.CalledByNative;
import org.chromium.base.annotations.JNINamespace;
import org.chromium.components.translate.TranslateMenu;

/**
 * Bridge for TranslateCompactInfoBar test methods invoked from native.
 */
@JNINamespace("weblayer")
public class TranslateTestBridge {
    TranslateTestBridge() {}

    // Selects the tab corresponding to |actionType| to simulate the user pressing on this tab.
    @CalledByNative
    private static void selectTab(TranslateCompactInfoBar infobar, int actionType) {
        infobar.selectTabForTesting(actionType);
    }

    @CalledByNative
    // Simulates a click of the overflow menu item for "never translate this language."
    private static void clickNeverTranslateLanguageMenuItem(TranslateCompactInfoBar infobar) {
        infobar.onOverflowMenuItemClicked(TranslateMenu.ID_OVERFLOW_NEVER_LANGUAGE);
    }

    @CalledByNative
    // Simulates a click of the overflow menu item for "never translate this site."
    private static void clickNeverTranslateSiteMenuItem(TranslateCompactInfoBar infobar) {
        infobar.onOverflowMenuItemClicked(TranslateMenu.ID_OVERFLOW_NEVER_SITE);
    }
}
